aptos-move/block-executor/src/limit_processor.rs

```diff
@@ -83,6 +89,32 @@ impl<T: Transaction> BlockGasLimitProcessor<T> {
         }
     }
 
+    pub(crate) fn recompute_acc_eff_block_gas_on_module_rw_conflict(&mut self) {
```

Comment:
let's call it: update_on_first_module_rw_conflict 

and let's also assert inside that module_rw_conflict is false